src/nisarqa/raster_classes.py

```diff
@@ -626,34 +656,127 @@ def decimate_raster_array_to_square_pixels(
     raster_obj: RadarRaster | GeoRaster,
 ) -> np.ndarray:
     """
-    Get *Raster's data array, decimated to square pixels in X and Y direction.
+    Decimate *Raster's data array to approx. square pixels in X and Y direction.
+
+    Parameters
+    ----------
+    raster_obj : RadarRaster or GeoRaster
+        *Raster object whose .data attribute will be read into memory
+        and decimated along the first two dimensions to approx. square pixels.
+
+    Returns
+    -------
+    out : numpy.ndarray
+        Copy of raster_obj.data array that has been decimated along the
+        first two dimensions to have approx. square pixels.
+    """
+    # Decimate to square pixels.
+    return decimate_array_to_square_pixels(
+        arr=raster_obj.data[...],
+        y_axis_spacing=raster_obj.y_axis_spacing,
+        x_axis_spacing=raster_obj.x_axis_spacing,
+    )
+
+
+def decimate_raster_array_to_square_pixels_with_strides(
+    raster_obj: RadarRaster | GeoRaster,
+) -> tuple[np.ndarray, int, int]:
+    """
+    Decimate *Raster's data array to square pixels and also return strides.
+
+    Pixels will be approx. square in the X and Y direction.
 
     Parameters
     ----------
     raster_obj : RadarRaster or GeoRaster
         *Raster object whose .data attribute will be read into memory
-        and decimated along the first two dimensions to square pixels.
+        and decimated along the first two dimensions to approx. square pixels.
 
     Returns
     -------
     out : numpy.ndarray
         Copy of raster_obj.data array that has been decimated along the
-        first two dimensions to have square pixels.
+        first two dimensions to have approx. square pixels.
+    ky, kx : int
+        The stride used for performing decimation in the X and Y directions,
+        respectively.
+    """
+    # Decimate to square pixels.
+    return decimate_array_to_square_pixels_with_strides(
+        arr=raster_obj.data[...],
```

Comment:
`raster_obj.data` has type `numpy.typing.ArrayLike`, which doesn't guarantee that we can perform this operation (`raster_obj.data[...]`). All we know is that we can convert it to a NumPy array.

Let's ensure that this always returns a copy, because functions sometimes return a copy and sometimes return a view can be tricky to use safely.

```suggestion
        arr=np.asarray(raster_obj.data, copy=True),
```